app/[lang]/page.tsx

```diff
@@ -0,0 +1,104 @@
+import { Locale } from "@/i18n.config";
+import Image from "next/image";
+import { getDictionary } from "@/lib/dictionary";
+
+export default async function Home({
+    params: { lang },
+}: {
+    params: { lang: Locale };
+}) {
+    const { page } = await getDictionary(lang);
```

Comment:
Potremmo facilitare la vita per i client component con un provider che si prenda la lang nel layout e poi esponga un hook direttamente con il dictionary corretto, che ne pensi?

Reply:
Ho implementato il Context Provider, però ho un dubbio sul cambio lingua, volevo passare il valore di params.lang al provider, ma al momento del mounting è un oggetto vuoto e mi trovo costretto ad usare la funzione 'changeLanguage' con lo useEffect in ogni punto in cui ho bisogno di usare la localizzazione

Reply:
Ho aggiunto il provider con l'hook e spostato il layout nella folder [lang], prima non era corretto